test/JDBC/input/BABEL-4271-vu-verify.sql

```diff
@@ -0,0 +1,30 @@
+-- Test to check ESCAPE null case (ESCAPE null means no ESCAPE char used)
+select 1 where 'ABCD' LIKE 'AB[C]D' ESCAPE '';
+go
+select 1 where 'cbc' LIKE '[c-a]bc' ESCAPE '';
+go
+select 1 where 'abc' LIKE '[0-a]bc' ESCAPE '';
+go
+select 1 where 'abc' LIKE '[abc]bc' ESCAPE '';
+go
+select 1 where 'abc' LIKE '[a-c]bc' ESCAPE '';
+go
+select 1 where 'bbc' LIKE '[a-c]bc' ESCAPE '';
+go
+select a, b from testvikasprj where testvikasprj.a LIKE testvikasprj.b ESCAPE '';
+go
+-- Test to check ESCAPE null case (ESCAPE null means no ESCAPE char used)
+select 1 where 'ABCD' LIKE 'AB[C]D' ESCAPE null;
+go
+select 1 where 'cbc' LIKE '[c-a]bc' ESCAPE null;
+go
+select 1 where 'abc' LIKE '[0-a]bc' ESCAPE null;
+go
+select 1 where 'abc' LIKE '[abc]bc' ESCAPE null;
+go
+select 1 where 'abc' LIKE '[a-c]bc' ESCAPE null;
+go
+select 1 where 'bbc' LIKE '[a-c]bc' ESCAPE null;
+go
+select a, b from testvikasprj where testvikasprj.a LIKE testvikasprj.b ESCAPE null;
```

Comment:
Let us have case like:

SELECT a, 'abc' from t1 where t1.a LIKE 'abc' ESCAPE null;
SELECT a, '' from t1 where t1.a LIKE '' ESCAPE null;
SELECT a, '' from t1 where t1.a LIKE '' ESCAPE '';
SELECT a, 'abc' from t1 where t1.a LIKE '' ESCAPE '';
So please try to think of such different scenarios to make sure that we are generating expected output.

Reply:
Added more test cases !